### ISSUE TITLE
Fold in functionality of Drop My Torch

### ISSUE THREAD

**Issue by derekstucki** [League-of-Foundry-Developers/torch#12]
My module, [Drop My Torch](https://gitlab.com/derekstucki/drop-my-torch), has partially overlapping functionality with Torch. Please contact me at derekstucki@gmail.com if you'd be interested in me rolling the DMT functionality missing from Torch into it.

**Comment by lupestro** [comment]
You may like what I implemented this weekend. I'll see if I can pull together a vid of it and send it along for your critique.

**Comment by lupestro** [comment]
I just published v2.0.0 for Foundry V10 and would be interested in your take on what I've done. I think I covered the bases but let me know if you see anything I missed. (For now, only run FoundryVTT 10 on a test server. It is still *very much* a work in progress and various systems and modules may corrupt data or cause hangs that require restarts. This will improve sharply over the coming few weeks.)

**Comment by steveculshaw** [comment]
I've installed v2.1.3 and I can't see anyway to "drop my torch" :-(

Am I missing something?

**Comment by lupestro** [comment]
So for dropping it, you mean placing it as a standalone object? I had thought of "Drop My Torch" as primarily adding light sources, but I somehow missed the implications of the name.

As for setting torches loose from the character, we haven't done that. All the lighting is "self" lighting as yet. I've thought about what it would take to implement a drop button on the context menu, creating an actor token that carries the light source. That then leaves the follow-on tasks of picking it up again or any party member extinguishing it.

I agree it would be useful and improve immersion. I just haven't gotten there yet and have spent a couple of months absorbing the implications of the new possibilities in the work I recently did. So I will concede that and add the enhancement tag to this ticket so I don't overlook it.

**Comment by steveculshaw** [comment]
Yeah, I was thinking of the use case, e.g., the character has a torch in hand, sees a monster, wants to drop the torch, and go into melee with their two handed sword.

Thanks for sharing the module

**Comment by lupestro** [comment]
A torch will light and stay lit where the player is standing when it is in inventory. We're not checking any further system bits like whether it is in hand, in a bag, etc.  When they drop the torch, is it extinguished? You can just click the button to turn it off before you swing your big honking' sword. :) What we can't do for you, at least at present, is to model your character  dancing away from a torch lying on the ground and shedding light from there. A feature worth having, but not a feature we have.

Philosophically, Foundry VTT is a system to use for playing a game, not necessarily a deeply enforced simulation of every game mechanic, and we're very much in keeping with that. Aside from the one consideration mentioned, I think there's nothing that will prevent you from handling the scenario you describe purely as "Theatre of the Mind".

**Comment by steveculshaw** [comment]
@apoapostolov Sounds promising suggestion. Ta
